### PR TITLE
Start adding unit tests to the backend

### PR DESCRIPTION
**Pull request by knorrium**
Supersedes #1903

### CODE REVIEW

**Comment by wiz** [comment]
please rebase onto current master

**Comment by wiz** [comment]
@knorrium <3

**Comment by knorrium** [comment]
@wiz <3

File: backend/package-lock.json [hunk 1]
```diff
@@ -23,94 +25,88 @@
...
-        "import-fresh": "^3.2.1",
-        "js-yaml": "^4.1.0",
-        "minimatch": "^3.1.2",
-        "strip-json-comments": "^3.1.1"
+        "@jridgewell/gen-mapping": "^0.1.0",
+        "@jridgewell/trace-mapping": "^0.3.9"
       },
       "engines": {
-        "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
+        "node": ">=6.0.0"
```
Comment (wiz) [inline backend/package-lock.json:55]: wut

**Review by wiz — APPROVED** [review]
utACK - this PR adds over 9000 lines of npm deps, but everything except babel is in devDeps, so I guess that's ok

**Pull request merged**